protobuf/flow/entities/block_execution_data.proto

```diff
@@ -8,33 +8,98 @@ option java_package = "org.onflow.protobuf.entities";
 import "flow/entities/event.proto";
 import "flow/entities/transaction.proto";
 
+// BlockExecutionData represents the collection of data produced while execiting the block.
 message BlockExecutionData {
+  // Block ID of the block that was executed.
   bytes block_id = 1;
+
+  // Ordered list of ChunkExecutionData produced while executing the block.
+  //
+  // Note: there will be one ChunkExecutionData per collection in the block, plus one for the
+  // service chunk. The service chunk is executed last and is always the last chunk in the list.
   repeated ChunkExecutionData chunk_execution_data = 2;
 }
 
+// ChunkExecutionData represents the collection of data produced while executing a chunk.
 message ChunkExecutionData {
+  // Ordered list of transactions included in the collection that was executed in the chunk.
   ExecutionDataCollection collection = 1;
+
+  // Events emitted by transactions in the collection.
+  //
+  // Note: events listed in the last ChunkExecutionData in the BlockExecutionData were emitted by
+  // the service transaction. Some, but not all, of these events are service events.
   repeated Event events = 2;
+
+  // TrieUpdate produced by executing the collection.
+  //
+  // TrieUpdates contain a list of registers that were modified during chunk execution. The value
+  // included is the new value of the register.
   TrieUpdate trieUpdate = 3;
+
+  // Transaction results produced by executing the collection.
+  //
+  // Note: these are not the same type of results returned by other RPCs. These results are sepcific
+  // to execution data. The most notable difference is they only include a boolean value to indicate
+  // whether or not an error was encountered during execution, not the error itself.
+  repeated ExecutionDataTransactionResult transaction_results = 4;
 }
 
+// ExecutionDataCollection represents the collection of transactions that were executed within a chunk.
+//
+// Note: this is not the same type as the entities.Collection.
 message ExecutionDataCollection {
+  // List of transactions included in the collection.
   repeated Transaction transactions = 1;
 }
 
+// TrieUpdate produced by executing the collection.
+//
+// TrieUpdates contain a list of registers that were modified during chunk execution. The value
+// included is the new value of the register.
 message TrieUpdate {
+  // RootHash is the root hash of the trie that included the contained updates.
```

Comment:
```suggestion
  // RootHash is the root hash of the trie that before the update is applied.
```

Reply:
See the implementation here, where a ledger update [is created with State as baseState](https://github.com/onflow/flow-go/blob/v0.31/engine/execution/state/state.go#L237). (baseState is the root hash of the trie before the update).

[And here the ledger update is converted into trie update with state copied to RootHash field](https://github.com/onflow/flow-go/blob/v0.31/ledger/complete/ledger.go#L206)